### ISSUE TITLE
Show differences in column order between dataframes

### ISSUE THREAD

**Issue by whtns** [gowerc/diffdf#32]
Sorry if I misunderstood the `diffdf` function. I'm just surprised by the behavior when I shuffle column names, ex. 

```
mtcars1 <- tibble::as_tibble(mtcars, rownames = "model")
mtcars2 <- mtcars1[,sample(colnames(mtcars1))]
diffdf::diffdf(mtcars1, mtcars2, keys = "model")
#> No issues were found!
```

**Comment by kieranjmartin** [comment]
Hi,

This is the intended behaviour. The goal of diffdf is to match columns on
name. This allows it to identify missing or additional columns
intelligently.

We have thought about having a strict column ordering arguments, but it
would require significant thought. If I have a data frame with columns a
and b, and columns a, c and b, using strict ordering I'm not sure how
helpful it is for the user to get a breakdown of the differences in the
second column, as we would expect c and b to look very different.

I think if we were to have a strict ordering argument, we would only have a
break down if the column matched name and position. Do you have any
thoughts on this?

**Comment by whtns** [comment]
Hmm, I think maybe just if this was clear on the readme? I obviously did not think through the comparison process in detail. If I had read a quick note before installing, I would use the package in its intended manner. Thanks!

**Comment by gowerc** [comment]
Hi,  @whtns  & @kieranjmartin ,

I am currently looking at implementing this.   My current thoughts are to disable this comparison by default then offer the option to turn it on with either checking prior to removing non-matching columns or post non-matching columns

something like 

```
check_column_order = c( "no", "prior", "post")
```

i.e. if we have  data.frame.1 with columns   a , b & c and data.frame.2 with columns a & c then under  `post` they would pass this check however under `prior` they would fail this check.

**Comment by kieranjmartin** [comment]
I think this is a pretty confusing implementation... To understand it from the user point of view you need to understand what diffdf actually does. I'm inclined to just utilise prior, and have strict column order as an option, as I suggest above. If you have columns with different names in two data frames you want to compare then I'd rather you just rename them

**Comment by gowerc** [comment]
Yerr, thinking about this some more  I defo think this is a worthwhile check including.  There are several (albeit niche) situations where order is important for example if you convert it to a matrix and then do matrix multiplication for calculating  least squares / empirical means estimates. 

Agreed my previous approach was too complicated,  am now thinking we do just a simple version of this were we compare column order after removing all non-matching columns  i.e.

dataset 1  =   Col A  ,  Col B ,  Col C
dataset 2 = Col A, Col D , Col C

would get reduced to comparing:
dataset 1 = Col A, Col C
dataset 2 =  Col A , Col C

and would thus pass

**Comment by gowerc** [comment]
@kieranjmartin

**Comment by kieranjmartin** [comment]
Is this the same as #28?

**Comment by gowerc** [comment]
Yes it is,  given the extended conversation here I would be up for closing #28
